File: leet_SLWD_1658_reduce_x.go
func minOperations(nums []int, x int) int {
    tt := 0
	for _, num := range nums {
		tt += num
	}
	len := len(nums)
	window := -1
	temp := 0
	R := -1
	L := 0
	for R < len - 1 {
		R++
		if R < len {
			temp += nums[R]
		}
		for temp > tt - x && L <= R {
			temp -= nums[L]
			L++
		}
		if temp == tt - x {
			if window < R - L + 1 {
				window = R - L + 1
			}
		}
	}
	if window == -1 {
		return -1
	}
	return len - window
}

File: leet_SLWD_1658_reduce_x.ts
function minOperations(nums: number[], x: number): number {
    let tt = nums.reduce((acc, res) => acc + res, 0)
    let len = nums.length
    let window = -1
    let temp = 0
    let R = -1
    let L = 0
    while (++R < len) {
        temp += nums[R]
        while (temp > tt - x && L <= R) {
            temp -= nums[L++]
        }
        if (temp == tt - x) {
            window = Math.max(window, R - L + 1);
        }
    }
    return window === -1 ? -1 : len - window
};

